applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -219,160 +247,142 @@ void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::GetFirstDerivativesVector(Vec
-    const IndexType rNumNodesNeighbour = rNeighbourGeom.size();
-    const IndexType NumGPointsNeighbour = rNeighbourGeom.IntegrationPointsNumber(rIntegrationMethodNeighbour);
+    const GeometryData::IntegrationMethod integration_method_neighbour = rNeighbourElement.GetIntegrationMethod();
+    const GeometryType& r_neighbour_geom = rNeighbourElement.GetGeometry();
+    const IndexType r_num_nodes_neighbour = r_neighbour_geom.size();
+    const IndexType num_g_points_neighbour = r_neighbour_geom.IntegrationPointsNumber(integration_method_neighbour);
 
-    Matrix rExtrapolationMatrix = ZeroMatrix(rNumNodesNeighbour, NumGPointsNeighbour);
+    Matrix extrapolation_matrix = ZeroMatrix(r_num_nodes_neighbour, num_g_points_neighbour);
 
     // Calculate extrapolation matrix for 2d elements
-    if (TDim == 2)
+    if constexpr (TDim == 2)
     {
-        if (rNumNodesNeighbour == 3)
+        if (r_num_nodes_neighbour == 3)
         {
-            GeoElementUtilities::CalculateExtrapolationMatrixTriangle(rExtrapolationMatrix, rIntegrationMethodNeighbour);
-            return rExtrapolationMatrix;
+            GeoElementUtilities::CalculateExtrapolationMatrixTriangle(extrapolation_matrix, integration_method_neighbour);
+            return extrapolation_matrix;
         }
-        if (rNumNodesNeighbour == 4)
+        if (r_num_nodes_neighbour == 4)
         {
-            GeoElementUtilities::CalculateExtrapolationMatrixQuad(rExtrapolationMatrix, rIntegrationMethodNeighbour);
-            return rExtrapolationMatrix;
+            GeoElementUtilities::CalculateExtrapolationMatrixQuad(extrapolation_matrix, integration_method_neighbour);
+            return extrapolation_matrix;
         }
        
     }
     // Calculate extrapolation matrix for 3d elements
-    if (TDim == 3)
+    if constexpr (TDim == 3)
     {
-        if (rNumNodesNeighbour == 4)
+        if (r_num_nodes_neighbour == 4)
         {
-            GeoElementUtilities::CalculateExtrapolationMatrixTetra(rExtrapolationMatrix, rIntegrationMethodNeighbour);
-            return rExtrapolationMatrix;
+            GeoElementUtilities::CalculateExtrapolationMatrixTetra(extrapolation_matrix, integration_method_neighbour);
+            return extrapolation_matrix;
         }
-        if (rNumNodesNeighbour == 8)
+        if (r_num_nodes_neighbour == 8)
         {
-            GeoElementUtilities::CalculateExtrapolationMatrixHexa(rExtrapolationMatrix, rIntegrationMethodNeighbour);
-            return rExtrapolationMatrix;
+            GeoElementUtilities::CalculateExtrapolationMatrixHexa(extrapolation_matrix, integration_method_neighbour);
+            return extrapolation_matrix;
         }
 
     }
 
     // if no extrapolation matrix is implemented, take average values at gauss points
-    double averaging_factor = 1 / NumGPointsNeighbour;
-    for (unsigned int node = 0; node < rNumNodesNeighbour; ++node)
+    const double averaging_factor = 1 / num_g_points_neighbour;
```

Comment:
Please be aware that with this expression the result will be truncated before the conversion to `double` is carried out. For instance, if `num_g_points_neighbour` equals 4, `averaging_factor` will be equal to 0.0, whereas you had probably expected 0.25. Here it is sufficient to replace 1 by 1.0.